pkg/driver/controllerServer.go

```diff
@@ -323,16 +325,31 @@ func (s *ControllerServer) ControllerExpandVolume(ctx context.Context, req *csi.
     }
     nsProvider := resolveResp.nsProvider
 
-    l.Infof("expanding volume %+v to %+v bytes", volumePath, capacityBytes)
-    err = nsProvider.UpdateVolume(volumePath, ns.UpdateVolumeParams{
-        VolumeSize: capacityBytes,
-    })
+    // Check if volume was not already expanded
+    l.Debugf("Checking volume %s size", volumePath)
+    volInfo, err := nsProvider.GetVolume(volumePath)
     if err != nil {
-        return nil, fmt.Errorf("Failed to expand volume volume %s: %s", volumePath, err)
+        return nil, err
+    }
+    currentSize := volInfo.VolumeSize
+    l.Debugf("Current size of volume %s = %+v", volumePath, currentSize)
+
+    if currentSize < capacityBytes {
+        l.Infof("expanding volume %+v to %+v bytes", volumePath, capacityBytes)
+        err = nsProvider.UpdateVolume(volumePath, ns.UpdateVolumeParams{
+            VolumeSize: capacityBytes,
+        })
+        if err != nil {
+            return nil, fmt.Errorf("Failed to expand volume volume %s: %s", volumePath, err)
```

Comment:
typo error: two times volume is printed.

Reply:
will fix, thanks